lib/ansible/modules/source_control/github_organization.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+module: github_organization
+short_description: Create/Rename a GitHub organization.
+description:
+    - Create or Rename a GitHub organization.
+version_added: "2.5"
+options:
+  state:
+    description:
+      -  Whether to create or rename an organization
+    choices: ['present', 'rename']
+    default: 'present'
+  name:
+    description:
+      - The name of the organization to create or rename
+    required: True
+  owner:
+    description:
+      - The owner of the organization that is being created.
+        The owner is required when I(state=present)
+  rename_to:
```

Comment:
no need for state=rename ... if rename_to is not None ... you want to rename.

Reply:
I would go one step further and use id as unique identifier and change name if it is different, avoiding rename_to/state=rename altogether

Reply:
Thanks, for the feedback, but when I went to implement it in that way, it does not hold together as nicely as it might first appear.

So we now have an `id` parameter that is the organization to create, you would typically not provide a `name` for creation, but you could do so.

If `id` and `name` are both provided, then we check if they are the same value.
If they are a the same, it is effectively a *create*, and will create the organization if it does not exist, otherwise doing nothing.

If they differ, then we have a *rename* and the `id` is used for the source organization to be renamed to `name` value. This is where we have the issue, as now effectively the old `id` no longer exists (as it was changed to `name`) if you re-run the playbook/action.
I fail the *rename* attempt if the source organization cannot be found, which I think is a requirement of that type of action, as *changed: False* does not feel right if you cannot find the thing to *rename*, as it is misleading (at least from my perspective).

I am on the fence about possibly disabling the *rename* feature, but I know that I would have used it in the past (which was why I added it).

Reply:
@bcoca Are you happy if I only refactor to the first suggestion due to what I commented on above?

Reply:
@bcoca friendly re-ping